doc/WebSite/Multi-Lingual-Entities.md

```diff
@@ -0,0 +1,160 @@
+
+A sample Dto class for sample Product entity above would be;
+
+```c#
+public class ProductListDto
+{
+    // Mapped from Product.Price
+    public decimal Price { get; set; }
+
+    // Mapped from ProductTranslation.Name
+    public string Name { get; set; }
+}
+```
+
+And it's mapping configuration is;
+
+```c#
+Configuration.Modules.AbpAutoMapper().Configurators.Add(configuration =>
+{
+	CustomDtoMapper.CreateMappings(configuration, new MultiLingualMapContext(
+		IocManager.Resolve<ISettingManager>()
+	));
+});
+
+internal static class CustomDtoMapper
+{
+	public static void CreateMappings(IMapperConfigurationExpression configuration, MultiLingualMapContext context)
+	{
+		configuration.CreateMultiLingualMap<Product, ProductTranslation, ProductListDto>(context);
+	}
+}
+```
+
+SettingManager is required to find default language setting when mapping a multi lingual entity to a Dto class. 
+
+In some cases like editing a multi lingual entity on the UI, all translations may be needed in the Dto class. In such cases, the Dto classes can be defined like below and [Object-To-Object-Mapping](Object-To-Object-Mapping.md) can be used.
+
+```c#
+[AutoMap(typeof(Product))]
+public class ProductDto
+{
+	public decimal Price { get; set; }
+    
+    public List<ProductTranslationDto> Translations {get; set;}
+}
+```
+
+```c#
+[AutoMap(typeof(ProductTranslation))]
+public class ProductTranslationDto
+{
+    public string Name { get; set; }
+}
+```
+### Crud Operations
+
+#### Creating a MultiLingual Entity with Translation(s) 
+
+A Dto class like the below one can be used for creating a Multi-Lingual entity with it's translations.
+
+```c#
+[AutoMap(typeof(Product))]
+public class ProductDto
+{
+	public decimal Price { get; set; }
+
+	public ICollection<ProductTranslationDto> Translations { get; set; }
+}
+```
+After defining such a Dto class, we can use it in our application service to create a Multi-Lingual entity.
+
+```c#
+public class ProductAppService : ApplicationService, IProductAppService
+{
+	private readonly IRepository<Product> _productRepository;
+
+	public ProductAppService(IRepository<Product> productRepository)
+	{
+		_productRepository = productRepository;
+	}
+	
+	public async Task CreateProduct(ProductDto input)
+	{
+		var product = ObjectMapper.Map<Product>(input);
+		await _productRepository.InsertAsync(product);
+	}
+}
+```
+#### Updating a Multi-Lingual Entity with Translation(s)
+
+We can use similar Dto class for updating our Multi-Lingual entity. A sample application service method for update operation can be defined like below;
+
+```c#
+public async Task UpdateProduct(ProductDto input)
+{
+	var product = await _productRepository.GetAsync(input.Id);
+	_productRepository.EnsureCollectionLoaded(product, p => p.Translations);
```

Comment:
Use `_productRepository.GetAll().Include(...)` instead of using `EnsureCollectionLoaded`.